### PR TITLE
Add the country of origin wizard step

### PR DESCRIPTION
**Pull request by tavy88**
### Jira link

HOTT-408

### What?

Add the country of origin wizard step

Progressive enhancement is ensured.

Also add .env.*.local files to gitignore.

### Why?

Users can now use the autocomplete field to pick
the country their products are coming from.

It's a good practice to keep secrets that will never be committed in the .env.*.local files.

### CODE REVIEW

File: app/controllers/wizard/steps/country_of_origin_controller.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    class CountryOfOriginController < BaseController
+      def show
+        @country = Wizard::Steps::CountryOfOrigin.new(session)
+      end
+
+      def create
+        @country = Wizard::Steps::CountryOfOrigin.new(session, permitted_params)
+
+        if @country.valid?
+          @country.save
```
Comment (tavy88) [inline app/controllers/wizard/steps/country_of_origin_controller.rb:12]: Redirect location here will be added in the following tickets.

File: .gitignore [hunk 2]
```diff
@@ -30,4 +30,5 @@ yarn-debug.log*
 .yarn-integrity
 
 .env
+.env.*.local
```
Comment (willfish) [inline .gitignore:33]: Yay \0/

File: app/controllers/wizard/steps/country_of_origin_controller.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+module Wizard
+  module Steps
+    class CountryOfOriginController < BaseController
+      def show
+        @country = Wizard::Steps::CountryOfOrigin.new(session)
```
Comment (willfish) [inline app/controllers/wizard/steps/country_of_origin_controller.rb:5]: How do you feel about renaming this to step? Feels like it wouldn't collide and also highlights we're coding to an interface and not an implementation, etc

File: app/models/commodity.rb [hunk 4]
```diff
@@ -12,10 +12,10 @@ def description
 
   def response
     @response ||= begin
-                    commodity = Uktt::Commodity.new(options)
-                    commodity.retrieve
-                    commodity.response
-                  end
+      commodity = Uktt::Commodity.new(options)
```
Comment (willfish) [inline app/models/commodity.rb:15]: The linter told me to do this :man_shrugging:

File: app/models/wizard/steps/country_of_origin.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+      def geographical_area_id
+        super || session[STEP_ID]
+      end
+
+      def save
+        session[STEP_ID] = geographical_area_id
+      end
+
+      def self.options_for(service)
+        Country.list(service.to_sym)
```
Comment (willfish) [inline app/models/wizard/steps/country_of_origin.rb:19]: :gb: :romania: :tada:

File: app/views/wizard/steps/country_of_origin/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+          <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
+            <h1 class="govuk-fieldset__heading">
+              Which country are the goods dispatched from?
+            </h1>
+          </legend>
+          <div class="govuk-form-group ">
+            <div id="destination-hint" class="govuk-hint">
+              The duty you are charged may be dependent on the country of dispatch of the goods being imported.
+            </div>
+            <%= f.govuk_collection_select :geographical_area_id, Wizard::Steps::CountryOfOrigin.options_for(params[:service_choice]), :id, :name, options: { prompt: true }, label: { text: 'Enter the country of dispatch:' } %>
```
Comment (willfish) [inline app/views/wizard/steps/country_of_origin/show.html.erb:19]:
```suggestion
            <%= f.govuk_collection_select :geographical_area_id, Wizard::Steps::CountryOfOrigin.options_for(service_choice), :id, :name, options: { prompt: true }, label: { text: 'Enter the country of dispatch:' } %>
```

File: spec/features/country_of_origin_page_spec.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    click_on('Continue')
+  end
+
+  it 'does not store an empty geographical area id on the session' do
+    click_on('Continue')
+
+    expect(Capybara.current_session.driver.request.session.key?(Wizard::Steps::CountryOfOrigin::STEP_ID)).to be false
+  end
+
+  xit 'looses its session key when going back to the previous question' do
```
Comment (willfish) [inline spec/features/country_of_origin_page_spec.rb:27]:
```suggestion
  xit 'loses its session key when going back to the previous question' do
```

File: spec/features/import_destination_page_spec.rb [hunk 8]
```diff
@@ -28,7 +28,7 @@
     expect(Capybara.current_session.driver.request.session[Wizard::Steps::ImportDestination::STEP_ID]).to eq('2')
   end
 
-  it 'loses its session key when going back to the previous question' do
+  it 'looses its session key when going back to the previous question' do
```
Comment (willfish) [inline spec/features/import_destination_page_spec.rb:31]:
```suggestion
  it 'loses its session key when going back to the previous question' do
```

File: spec/fixtures/country.json [hunk 9]
```diff
@@ -0,0 +1,13 @@
+{
+  "data": [
+    {
+      "id": "NI",
+      "type": "geographical_area",
+      "attributes": {
+        "id": "NI",
+        "description": "Nicaragua",
+        "geographical_area_id": "NI"
```
Comment (willfish) [inline spec/fixtures/country.json:9]: :nicaragua:

File: yarn.lock [hunk 10]
```diff
@@ -6093,6 +6100,11 @@ postcss@^7.0.0, postcss@^7.0.1, postcss@^7.0.14, postcss@^7.0.17, postcss@^7.0.2
     source-map "^0.6.1"
     supports-color "^6.1.0"
 
+preact@^8.3.1:
```
Comment (willfish) [inline yarn.lock:6103]: Interesting :+1:

**Review by willfish — APPROVED** [review]

File: app/models/commodity.rb (hunk 4 above)
Comment (tavy88) [reply app/models/commodity.rb:15]: weird...

File: yarn.lock (hunk 10 above)
Comment (tavy88) [reply yarn.lock:6103]: Yeah, it comes with the component.

**Pull request merged**